### ISSUE TITLE
[Installer] Refine the environment backup and restoration

### ISSUE THREAD

**Issue by q82419** [WasmEdge/WasmEdge#1058]
### Current State

For the MacOS platforms, we only use the `.bash_profile` for the bash configuration.
But after the installation by the installer, the `.bashrc` and `.profile` are generated.
And then after the uninstallation, we will get `.bashrc.wasmedge_backup`, `.profile.wasmedge_backup` and the 0 size of `.bashrc` and `.profile`.

### Expected

Not to leave the trash after the uninstaller execution.
We should not generate the unused config files such as `.bashrc` and `.profile` on MacOS platforms.
As the comment in PR #1057 , I think we can follow the rules:

1. Search the existing bash config file and add the `. "$HOME/.wasmedge/env"` command into it.
2. If there's no config file, create one according to OS: `.bash_profile` for MacOS, `.bashrc` for Ubuntu, etc.
3. For uninstaller, only restore and backup the edited config file.

Further more, I think it's much better to edit the config file to remove the `. "$HOME/.wasmedge/env"` line when uninstalling.
Because the users may add or edit the config file after the WasmEdge installation. If we restore the config file, the settings added after we backup will also be restored. Therefore the users' environment is changed by the uninstaller without their acknowledgement.

## Environment

- Hardware Architecture: x86_64
- Operating system: MacOS 15

## Steps to Reproduce

Use the installer and uninstaller.

**Comment by SAtacker** [comment]
> We should not generate the unused config files such as .bashrc and .profile on MacOS platforms.

For this purpose, there is https://github.com/WasmEdge/WasmEdge/blob/706a42ccaf1b6d79ac8e65bd01e6e5aa5b3ceaaf/utils/install.sh#L590 which detects the shell configuration file.
https://github.com/WasmEdge/WasmEdge/blob/706a42ccaf1b6d79ac8e65bd01e6e5aa5b3ceaaf/utils/install.sh#L610-L616

The below
https://github.com/WasmEdge/WasmEdge/blob/706a42ccaf1b6d79ac8e65bd01e6e5aa5b3ceaaf/utils/install.sh#L593-L609
is added to forcibly make files whether they exist or not.

## TLDR; Remove 593-609 to achieve 2,3
- [X] 1. Search the existing bash config file and add the . "$HOME/.wasmedge/env" command into it.
- [ ] 2. If there's no config file, create one according to OS: .bash_profile for MacOS, .bashrc for Ubuntu, etc.
- [ ] 3. For uninstaller, only restore and backup the edited config file.

**Comment by SAtacker** [comment]
>Further more, I think it's much better to edit the config file to remove the . "$HOME/.wasmedge/env" line when uninstalling.
Because the users may add or edit the config file after the WasmEdge installation. If we restore the config file, the settings added after we backup will also be restored. Therefore the users' environment is changed by the uninstaller without their acknowledgement.

While running the uninstaller, we are simply removing that particular line that contains `. <WasmEdge_Installation_Path>`. While doing this, we also take a backup of that file so that in case of a mishap user can refer to the backup file with the extension `.wasmedge_backup`. Hence it does not matter if users modify their configuration files anytime before or after uninstallation as we only remove that line if found in their default configuration shell. 
I hope my understanding was correct. @q82419 
Working on the rest of the things mentioned.
Thanks.

**Comment by q82419** [comment]
Fixed.
